include/seqan3/range/container/small_string.hpp

```diff
@@ -309,6 +337,53 @@ class small_string : public small_vector<char, capacity_ + 1>
         return str();
     }
     //!\}
+
+    /*!\name Input/output
+     * \{
+     */
+
+    /*!\brief Formatted output for the seqan3::small_string.
+     * \param[in,out] os  The std::basic_ostream to write to.
+     * \param[in]     str The seqan3::small_string to read from.
+     * \returns `os`.
+     */
+    friend std::ostream & operator<<(std::ostream & os, small_string const & str)
+    {
+        os << str.str();
+        return os;
+    }
+
+    /*!\brief Formatted input for the seqan3::small_string.
+     * \param[in,out] is  The std::basic_istream to read from.
+     * \param[out]    str The seqan3::small_string to write to.
+     * \returns `is`.
+     */
+    friend std::istream & operator>>(std::istream & is, small_string & str)
+    {
+        // Check if stream is ok and skip leading whitespaces.
+        std::istream::sentry s(is);
+        if (s)
+        {
+            str.erase(); // clear the string
+            std::streamsize num_char = (is.width() > 0) ? is.width() : str.max_size();
```

Comment:
take the maximum of width and str.max_size so you don't exceed the capacity and document the behaviour (only comsume at most str.capacity characters from stream).